routes/product/helpers.js

```diff
@@ -19,6 +19,23 @@ export default {
             return res.json({ error, success: false });
         }
     },
+    //clean up invalid products and producers
+    cleanProductsWithInvalidProducerId: async (req, res) => {
+        try {
+            const data = await Product.find();
+            const invalidProducer = data.map(dataitem => {
+                if (!ObjectID.isValid(dataitem.producerId) || !ObjectID.isValid(dataitem._id)) {
```

Comment:
why are you doing this check here?

Reply:
We added some invalid producers and products previously when we started. their IDs are assigned manually then. Here, i am using this check to find them and delete them.
it supposed to be a one time feature but i allowed it for redundancy purposes